hoon/lib/jock.hoon

```diff
@@ -200,13 +208,18 @@
       name=term
   ==
 ::
-+$  jype-leaf
-  $%  [%atom p=jatom-type]
++$  base-jype-leaf
+  $%  [%atom p=jatom-type q=?(%.y %.n)]
       [%core p=core-body q=(unit jype)]
       [%limb p=(list jlimb)]
-      [%symbol p=jatom-type q=@]
       [%fork p=jype q=jype]
-      [%untyped ~]
+      [%none ~]
+  ==
+::
++$  jype-leaf
+  $%  base-jype-leaf
+      [%ring p=base-jype-leaf q=jype]
```

Comment:
Please use semantically meaningful names for these things. Ring is ok but font doesn't make sense. The general rule should be: if you have to explain to someone reading your code, "a font is really a _this_" you should just name it something more meaningful.

Reply:
Four letters align better because of gap discipline in Hoon; `%loop` is already a keyword.  `%self` would be deceptive (since it's only itself in the null case).  Still thinking about names.

Reply:
Semantic meaning trumps beauty or readability (within limits, shorter names are preferred) imo.

Reply:
Logan is correct here

Reply:
`%link` could work, or `%indx`